src/Aggregation/Aggregation.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+
+namespace Novaway\ElasticsearchClient\Aggregation;
+
+
+class Aggregation
+{
+    /** @var  string */
+    protected $category;
+    /** @var  string */
+    protected $name;
+    /** @var  string */
+    protected $field;
+    /** @var  array */
+    protected $options;
+
+    /**
+     * Aggregation constructor.
+     * @param $name         name of the key in results
+     * @param $category     category of aggregation : terms, max, avg ...
+     * @param $field        field on which the aggregation is done
+     * @param $options      extraneous parameters given to the aggregation, needed in some categories
+     */
+    public function __construct(string $name, string $category, string $field, array $options = [])
+    {
+        $this->category = $category;
+        $this->name = $name;
+        $this->field = $field;
+        $this->options = $options;
+    }
+
+    /**
+     * @return string
+     */
+    public function getCategory(): string
+    {
+        return $this->category;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    /**
+     * @return string
+     */
+    public function getField(): string
+    {
+        return $this->field;
+    }
+
+    /**
+     * @return array
+     */
+    public function getOptions(): array
+    {
+        return $this->options;
+    }
+
+    public function formatParameters(): array
```

Comment:
IMO formatting parameters is not the value object responsability.

It should be done where formatting is necessary or extract in another object.

Reply:
maybe it would be better named as getParameters.

or even done in the constructor, as there is no real point to differentiate $fields and $options past the objects instanciation , and saved in a new  $parameters attribute

Reply:
Yes, `getParameters` could be a better name and avoid confusion and should be enough for this PR